File: Karum/src/test/java/utility/WebScrap.java
package test.java.utility;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.io.File;

public class WebScrap {
    private WebDriver webdriver;
    private WebDriverWait webdriverwait;

    public WebScrap() {
        File file = new File("chromedriver.exe");
        System.setProperty("webdriver.chrome.driver", file.getAbsolutePath());
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--headless");
        webdriver = new ChromeDriver(options);
        webdriverwait = new WebDriverWait(webdriver, 120);
    }

    public String RecoverDataElementPage(
            String url, By pathElementData, String attributeToRecover, String Iframe) {

        String returnData = "";

        try {
            webdriver.navigate().to(url);
            Thread.sleep(1000);
            if(!Iframe.isEmpty()) {
                webdriverwait.until(ExpectedConditions.frameToBeAvailableAndSwitchToIt(By.id(Iframe)));
            }

            webdriverwait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(pathElementData));
            WebElement pageElement = webdriver.findElement(pathElementData);

            if(attributeToRecover.isEmpty()) {
                returnData = pageElement.getText();
            }
            else {
                returnData = pageElement.getAttribute(attributeToRecover);
            }
        }
        catch (Exception exception) {
            System.out.println(exception.getMessage());
        }

        return returnData;
    }

    public String RecoverDataElementPage(
            String url, By pathElementData, String attributeToRecover) {
        return RecoverDataElementPage(url, pathElementData, attributeToRecover, "");
    }

    public void waitElementExist(String url, By locator) {
        webdriver.navigate().to(url);
        webdriverwait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(locator));
    }

    public void KillSession() {
        webdriver.close();
        webdriver.quit();
    }
}

File: KarumMobileTest/KarumMobileTest/utility/WebScrap.cs
namespace utility
{
    using OpenQA.Selenium;
    using OpenQA.Selenium.Chrome;
    using OpenQA.Selenium.Support.UI;
    using System;
    using System.Threading;

    public class WebScrap 
    {
        private IWebDriver webdriver;
        private WebDriverWait webdriverwait;

        /// <summary>
        /// Contructor Create a Instance of Chrome Driver for WebScraping
        /// </summary>
        public WebScrap() 
        {
            ChromeOptions options = new ChromeOptions();
            options.AddArguments("--headless");
            options.AddArguments("--window-size=1920,1080");
            webdriver = new ChromeDriver(options);
            webdriverwait = new WebDriverWait(webdriver, TimeSpan.FromMinutes(5));
        }

        /// <summary>
        /// Recover the text/Attribute value from a Element
        /// </summary>
        /// <param name="url">URL webpage</param>
        /// <param name="pathElementData">Element By object</param>
        /// <param name="attributeToRecover">the name of the element Attribute what value is going to be recover. If is Empty the element text is going to be recover</param>
        /// <param name="Iframe">In case is necesary to change Iframe. OPTIONAL</param>
        /// <returns>Element attribute value</returns>
        public string RecoverDataElementPage(
                string url, By pathElementData, string attributeToRecover, string Iframe = "") 
        {
            string returnData = "";

            try {
                webdriver.Navigate().GoToUrl(url);
                Thread.Sleep(1000);
                if (!Iframe.Equals(string.Empty)) {
                    webdriverwait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.FrameToBeAvailableAndSwitchToIt(By.Id(Iframe)));
                }

                webdriverwait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.VisibilityOfAllElementsLocatedBy(pathElementData));
                IWebElement pageElement = webdriver.FindElement(pathElementData);

                if (attributeToRecover.Equals(string.Empty)) 
                {
                    returnData = pageElement.Text;
                }
                else 
                {
                    returnData = pageElement.GetAttribute(attributeToRecover);
                }
            }
            
            catch (Exception exception) 
            {
                Console.WriteLine(exception.Message);
            }

            return returnData;
        }

        /// <summary>
        /// Wait until a element is visible on Screen
        /// </summary>
        /// <param name="url">URL webpage</param>
        /// <param name="locator">By to find the element</param>
        public void waitElementExist(string url, By locator) 
        {
            webdriver.Navigate().GoToUrl(url);
            webdriverwait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.VisibilityOfAllElementsLocatedBy(locator));
        }

        public void KillSession()
        {
            webdriver.Close();
            webdriver.Quit();
        }
    }
}